dataland-frontend/src/components/forms/parts/elements/basic/UploadDocumentsForm.vue

```diff
@@ -0,0 +1,166 @@
+import { CompanyReport } from "@clients/backend";
+import Keycloak from "keycloak-js";
+import {
+  calculateSha256HashFromFile,
+  DocumentToUpload,
+  isThereActuallyANewFileSelected,
+  removeFileTypeExtension,
+} from "@/utils/FileUploadUtils";
+
+export default defineComponent({
+  name: "UploadDocumentsForm",
+  components: {
+    PrimeButton,
+    FileUpload,
+  },
+  emits: ["documentsChanged"],
+  setup() {
+    return {
+      getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
+    };
+  },
+  data() {
+    return {
+      formsDatesFilesToUpload: [] as string[] | undefined,
+      formatBytesUserFriendly,
+      DOCUMENT_UPLOAD_MAX_FILE_SIZE_IN_BYTES: DOCUMENT_UPLOAD_MAX_FILE_SIZE_IN_BYTES,
+      documentsToUpload: [] as DocumentToUpload[],
+    };
+  },
+  props: {
+    referencedDocumentsForPrefill: {
+      type: Object as () => { [key: string]: CompanyReport },
+    },
+    moreThanOneDocumentAllowed: {
+      type: Boolean,
+      default: true,
+    },
+  },
+
+  methods: {
+    removeFileTypeExtension,
+    /**
+     * Handles selection of a file by the user. Only considers the file that was added last.
+     * The file is added to the documents that shall be uploaded, then the sha256 hashes are calculated and added
+     * to the respective files.
+     * @param event full event object containing the files
+     * @param event.files files
+     */
+    async handleFilesSelected(event: FileUploadSelectEvent): Promise<void> {
+      const selectedFilesByUser = event.files as File[];
+      if (isThereActuallyANewFileSelected(selectedFilesByUser, this.documentsToUpload)) {
+        const lastSelectedFile = selectedFilesByUser[selectedFilesByUser.length - 1];
+        const documentToUpload = { file: lastSelectedFile } as DocumentToUpload;
+        documentToUpload.reference = await calculateSha256HashFromFile(documentToUpload.file);
+        documentToUpload.fileNameWithoutSuffix = removeFileTypeExtension(documentToUpload.file.name);
+        this.documentsToUpload.push(documentToUpload);
+        this.emitDocumentsChangedEvent();
+      }
+    },
+    /**
+     * Emits event that selected documents changed
+     */
+    emitDocumentsChangedEvent() {
+      this.$emit("documentsChanged", this.documentsToUpload);
+    },
+
+    /**
+     * Remove document from files uploaded
+     * @param indexOfFileToRemove index number of the file to remove
+     * @param deleteCount the number of files to delete
+     */
+    removeDocumentFromDocumentsToUpload(indexOfFileToRemove: number, deleteCount = 1) {
+      (this.$refs.fileUpload.remove as (index: number) => void)(indexOfFileToRemove);
+      this.documentsToUpload.splice(indexOfFileToRemove, deleteCount);
+      this.emitDocumentsChangedEvent();
+    },
+
+    /**
+     * removes all documents at once, is invoked by a yes no field if it is resetted to "No"
+     */
+    removeAllDocuments() {
```

Comment:
Are these new functions or are these refactored from Upload Reports?

Reply:
Part of them are refactored. `removeAllDocuments` is new as well as ``prefillFileUpload`

Reply:
Can you shortly explain to me why the dummy files are necessary?

Reply:
The dummyFile is necessary because we want the FileUpload to behave like it would if had actually selected a file. In the background all of this depends on a `File[]` which the FileUpload Component fills and displays. Therefore we need to provide it with something of type `File`, which we do not get from the backend if we fetch the existing dataset in edit mode. So, instead of separately calling the document API to get the real document, we construct an empty dummy file to contain the referenced documentName, that then is displayed by the FileUpload as already uploaded.

Reply:
But why don't we need this in the eu taxonomy forms? Can we maybe have a short call to discuss this?

Reply:
Had the call and now i understand it.